### DISCUSSION TITLE
How to keep my account after uni email gets deleted?

### DISCUSSION THREAD

**Discussion by Nerocraft4** [community/community#54341] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello everyone! First time posting here. I have searched around the forum, found a bit of info, but I'm still not sure on how to proceed.

My account's primary e-mail is a uni-provided email. I'm going to graduate soon, and once I leave my uni I might lose access to its email. I have a secondary (private) email linked to my account, for security reasons.

Should I make my secondary email (my own) the primary? Will I lose the account if I don't do this swap before they delete my uni email? Will my activity and commits still show?

Thank you! I'm in no hurry, still have a few months, but I'd like to clear this up.

~ Nero

**Comment by airtower-luna** [comment, marked as answer]
> Should I make my secondary email (my own) the primary?

Yes, make sure the primary address is one that actually works and that you have access to.

> Will I lose the account if I don't do this swap before they delete my uni email?

Possibly. If you don't have 2FA enabled GitHub will send a code to your mail address during login, so without it you won't be able to log in.

> Will my activity and commits still show?

Yes, as long as you don't delete the university email completely from your account (just select other primary/secondary addresses). You can have as many addresses on your account as you want, and for recognizing your commits it doesn't matter if one doesn't work any more. You might want to use a current address for new commits after leaving university (adjust `user.email` in your Git config), but as long as the address is on your account the commits will be recognized as yours. :wink:

**Reply by Nerocraft4** [reply]
Thank you so much! Will do!
